dig.go

```diff
@@ -21,31 +21,40 @@
 package dig
 
 import (
-	"bytes"
 	"errors"
 	"fmt"
+	"math/rand"
 	"reflect"
 	"strconv"
 	"strings"
+	"time"
 )
 
 const (
 	_optionalTag = "optional"
 	_nameTag     = "name"
+	_groupTag    = "group"
 )
 
 // Unique identification of an object in the graph.
 type key struct {
-	t    reflect.Type
-	name string
+	t reflect.Type
+
+	// Only one of name or group will be set.
+	name  string
+	group string
 }
 
 // Option configures a Container. It's included for future functionality;
 // currently, there are no concrete implementations.
 type Option interface {
-	unimplemented()
+	applyOption(*Container)
```

Comment:
Just `apply`? Only curious - obviously fine as-is too.

Reply:
Unexported methods on interfaces from the same package are in the same
namespace.

If you have,

    type Option interface {
        apply(*Container)
    }
    
    type InvokeOption interface {
        apply(*invokeOptions)
    }
    
You can't have a single object implement both interfaces.

This is obviously not needed at this time but may be required in the future.